tests/trainers/test_gpt3_text-generation_chinese-base_ddp.py

```diff
@@ -0,0 +1,181 @@
+# Copyright (c) Alibaba, Inc. and its affiliates.
+import os
+import shutil
+import tempfile
+import unittest
+import torch
+from modelscope.metainfo import Trainers
+from modelscope.msdatasets import MsDataset
+from modelscope.pipelines import pipeline
+from modelscope.trainers import build_trainer
+from modelscope.utils.test_utils import DistributedTestCase, test_level
+from modelscope.utils.hub import snapshot_download, read_config
+from modelscope.utils.hub import read_config, Config
+from modelscope.utils.constant import Tasks
+import socket
+from modelscope.utils.torch_utils import _find_free_port
+
+
+
+@unittest.skipIf(not torch.cuda.is_available()
+                 or torch.cuda.device_count() <= 1, 'distributed unittest')
+class TestGPT3BaseTrainDDP(DistributedTestCase):
+
+    def setUp(self):
+        print(('Testing %s.%s' % (type(self).__name__, self._testMethodName)))
+
+        self.tmp_dir = tempfile.TemporaryDirectory().name
+        if not os.path.exists(self.tmp_dir):
+            os.makedirs(self.tmp_dir)
+
+        self.model_dir = snapshot_download('damo/nlp_gpt3_text-generation_chinese-base')
+        config:Config = read_config(os.path.join(self.model_dir, 'configuration.json'))
+        config.megatron.world_size = 4
+        config.megatron.tensor_model_parallel_size = 2
+        # d = w // p // t
+        config.megatron.data_parallel_size = 2
```

Comment:
whether the base model has megatron component

Reply:
change to use pytorch ddp.